packages/core/src/Cardano/Address/PointerAddress.ts

```diff
@@ -30,26 +30,27 @@ const variableLengthEncode = (val: number): Buffer => {
 };
 
 /**
- * Decodes a value previously encoded in a variable length array.
+ * Decodes a value previously encoded in a variable length array only
+ * up to Number.MAX_SAFE_INTEGER values before decoding to 'Infinity'.
  *
  * @param array The encoded value.
  * @returns the decoded value.
  */
 const variableLengthDecode = (array: Buffer): { value: number; bytesRead: number } => {
   let more = true;
-  let value = 0;
+  let value = 0n;
 
   let bytesRead = 0;
   while (more && bytesRead < array.length) {
     const b = array[bytesRead];
-    value <<= 7;
-    value |= b & 127;
+    value <<= 7n;
+    value |= BigInt(b & 127);
 
     more = (b & 128) !== 0;
     ++bytesRead;
   }
 
-  return { bytesRead, value };
+  return { bytesRead, value: value <= Number.MAX_SAFE_INTEGER ? Number(value) : Number.POSITIVE_INFINITY };
```

Comment:
We should add a test for round-tripping a Pointer address that uses a `slot>MAX_SAFE_INTEGER`. In order for that to work, we'll need to change [this](https://github.com/input-output-hk/cardano-js-sdk/blob/34f01fef6d07c7d209929f2a4a47c8dfcc556ac9/packages/core/src/Cardano/Address/Address.ts#L107) Pointer type to store a `bigint`, and save the full value when packing in order to reuse when unpacking